roles/validate_inventory/tasks/required_vars.yml

```diff
@@ -4,3 +4,23 @@
       - repo_root_path is defined
     fail_msg: repo_root_path is required for all playbooks to function correctly
   changed_when: False
+
+- name: Check kubeadmin_vault_password_file_path is defined
+  assert:
+    that:
+      - kubeadmin_vault_password_file_path is defined
+    fail_msg: kubeadmin_vault_password_file_path is a required variable
+  changed_when: False
+
+- name: Check kubeadmin_vault_password_file_path file exists
+  block:
+    - name: Stat kubeadmin_vault_password_file_path
+      stat:
+        path: "{{ kubeadmin_vault_password_file_path }}"
+      register: stat_kubeadmin_vault_password_file_path
+    - name: assert exists
+      assert:
+        that:
+          - stat_kubeadmin_vault_password_file_path.stat.exists
+        fail_msg: "kubeadmin_vault_password_file_path ({{ kubeadmin_vault_password_file_path }}) does not exist"
+      changed_when: False
```

Comment:
This can be simplified to a single assert task using a [path test](https://docs.ansible.com/ansible/latest/user_guide/playbooks_tests.html#testing-paths):
```yaml
- name: Check kubeadmin vault password file path is defined and the file exists
  assert:
    that:
      - kubeadmin_vault_password_file_path is defined
      - kubeadmin_vault_password_file_path is file
    fail_msg: "Kubeadmin Vault password must be stored in the location specified by the required variable 'kubeadmin_vault_password_file_path'."
  changed_when: False
```

Reply:
Good spot! I like the message as well :)